dropwizard-jersey/src/main/java/io/dropwizard/jersey/optional/OptionalLongMessageBodyWriter.java

```diff
@@ -36,7 +37,7 @@ public void writeTo(OptionalLong entity,
                         MultivaluedMap<String, Object> httpHeaders,
                         OutputStream entityStream) throws IOException {
         if (!entity.isPresent()) {
-            throw new NotFoundException();
+            throw new NoContentException("Empty content.");
```

Comment:
Not sure if I'm a fan of using this exception, as it looks like it forces the use of another exception mapper.

Have you tried doing just a `WebApplicationException` with the `Response.noContent().build()`?